openedx/features/offline_content/docs/001-mobile-offline-content-support.rst

```diff
@@ -0,0 +1,153 @@
+
+
+* **Implement of a mechanism for generating and storing on a server or external storage**: The course content should be pre-generated and saved to the storage for later download.
+    * **Render block fragment**: Implement a new standard XBlock view called `offline_view` which would generate user-agnostic fragments suitable for offline use. This view will avoid any dependence on student-specific state, focusing solely on content and settings.
+    * **Replace static and media**: Save static and media assets files used in block to temporary directory and replace their static paths with local paths.
+    * **Archive and store content**: Archive the generated content and store it on the server or external storage.
+* **Mechanism for updating the generated data**: When updating course blocks (namely when publishing) the content that has been changed should be re-generated.
+    * **Track course publishing events on CMS side**: Add a new signal `course_cache_updated` to be called after the course structure cache update in `update_course_in_cache_v2`. Add a signal that listens to `course_cache_updated` and starts block generation.
+    * **Update archive**: Check generated archive creation date and update it if less than course publishing date.
+* **Implement a Mobile Local Storage Mechanism**: Use the device's local storage to save course content for offline access.
+    * **Extend blocks API**: Add links to download blocks content and  where it is possible.
+* **Sync Mechanism**: Periodically synchronize local data with the server when the device is online.
+    * **Sync on app side**: On course outline screen, check if the course content is up to date and update it if necessary.
+    * **Sync user responses**: When the device is offline, save user responses locally and send them to the server when the device is online.
+* **Selective Download**: Allow users to choose specific content to download for offline use.
+* **Full Course Download**: Provide an option to download entire courses for offline access.
+
+Supported xBlocks in offline mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It was decided to include a fraction of Open edX xBlocks to be supported.
+The following list of blocks is currently planned to be added to the support:
+
+* **Common problems**:
+    * **Checkboxes** - full support
+    * **Dropdown** - full support
+    * **Multiple Choice** - full support
+    * **Numerical Input** - full support
+    * **Text Input** - full support
+    * **Checkboxes with Hints and Feedback** - partial support without Hints and Feedback
+    * **Dropdown with Hints and Feedback** - partial support without Hints and Feedback
+    * **Multiple Choice with Hints and Feedback** - partial support without Hints and Feedback
+    * **Numerical Input with Hints and Feedback** - partially supported without Hints and Feedback
+    * **Text Input with Hints and Feedback** - partially supported without Hints and Feedback
```

Comment:
```suggestion
    * **Text Input with Hints and Feedback** - partially supported without Hints and Feedback
    * **Blank Advanced Problems** - partially supported, without loncapa/python problems or multi-part problems
```

I presume that python-evaluated problems will not be supported. @stan-neichev can you confirm?